app/views/RoomInfoView/components/UserInfoIcon.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { Text } from 'react-native';
+import { BorderlessButton } from 'react-native-gesture-handler';
+
+import { CustomIcon, TIconsName } from '../../../containers/CustomIcon';
+import styles from '../styles';
+import { useTheme } from '../../../theme';
+import { useVideoConf } from '../../../lib/hooks/useVideoConf';
+import i18n from '../../../i18n';
+import { useAppSelector } from '../../../lib/hooks';
+import { compareServerVersion } from '../../../lib/methods/helpers';
+
+// TODO: change other icons on future
+export function UserInfoIcon({
+	danger,
+	iconName,
+	onPress,
+	label,
+	showIcon
+}: {
+	danger?: boolean;
+	iconName: TIconsName;
+	onPress?: (prop: any) => void;
+	label: string;
+	showIcon?: boolean;
+}): React.ReactElement | null {
+	const { colors } = useTheme();
+	const color = danger ? colors.dangerColor : colors.actionTintColor;
+
+	if (showIcon)
+		return (
+			<BorderlessButton testID={`room-info-view-${iconName}`} onPress={onPress} style={styles.roomButton}>
+				<CustomIcon name={iconName} size={30} color={color} />
+				<Text style={[styles.roomButtonText, { color }]}>{label}</Text>
+			</BorderlessButton>
+		);
+	return null;
+}
+
+export function CallIcon({ rid, isDirect }: { rid: string; isDirect: boolean }): React.ReactElement | null {
+	const { showCallOption, showInitCallActionSheet } = useVideoConf(rid);
+	const serverVersion = useAppSelector(state => state.server.version);
+	const greaterThanFive = compareServerVersion(serverVersion, 'greaterThanOrEqualTo', '5.0.0');
+
+	const showIcon = greaterThanFive ? showCallOption : showCallOption && isDirect;
+
+	return <UserInfoIcon onPress={showInitCallActionSheet} iconName='phone' label={i18n.t('Call')} showIcon={showIcon} />;
```

Comment:
Do we need to reuse or could we do something like this?

```tsx
const { showCallOption, showInitCallActionSheet } = useVideoConf(rid);
const serverVersion = useAppSelector(state => state.server.version);
const { colors } = useTheme();
const color = danger ? colors.dangerColor : colors.actionTintColor;
const greaterThanFive = compareServerVersion(serverVersion, 'greaterThanOrEqualTo', '5.0.0');

if (greaterThanFive ? showCallOption : showCallOption && isDirect) {
	return (
		<BorderlessButton>
			...
		</BorderlessButton>
	);
}

return null;
```

Looks a lot easier to read, but there's also that `danger` prop that is never used.
I don't know if you're seeking to use it in the future.
But then you should move it to a separate file, so I'm not sure.

Reply:
The button will be used for the other buttons on the screen.